### PR TITLE
 feat(sdk.deleteIdentityData): delete credentials, identity, and interactions

### PR DESCRIPTION
**Pull request by mnzaki**
This PR depends on changes that are yet to be merged in https://github.com/jolocom/sdk-storage-typeorm/pull/19
Unfortunate circular dependency means that to test this one needs to clone, build, and link, this branch https://github.com/jolocom/sdk-storage-typeorm/tree/feat/queryAndDelete

I propose merging the sdk-storage-typeorm PR, then updating here to get the tests passing

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -285,4 +289,20 @@ export class JolocomSDK {
...
+      { subject: did },
+      { issuer: did },
+    ])
+    await this.storage.delete.identity(did)
+    let query: InteractionQueryAttrs[] = []
+    identity.publicKeySection.forEach((pk) => {
+      let keyId = `${did}${pk.id}`
+      query.push({ initiator: keyId }, { responder: keyId })
+    })
+    await this.storage.delete.interactions(query)
```
Comment (clauxx) [inline src/index.ts:306]: In what case would we use different public keys for different interactions?

File: src/credentials.ts [hunk 2]
```diff
@@ -98,3 +105,111 @@ export class CredentialType {
     }
   }
 }
+
+export class CredentialKeeper
```
Comment (clauxx) [inline src/credentials.ts:109]: Should the `CredentialKeeper` also be responsible for the credential metadata?

Comment (clauxx) [inline src/credentials.ts:109]: Should the `CredentialKeeper` be responsible for the credential metadata as well. It's used in the `display` method, but we also need parts of `renderInfo` as well.

**Review by clauxx — APPROVED** [review]
LGTM (tested on the SmartWallet)

File: src/index.ts (hunk 1 above)
Comment (mnzaki) [reply src/index.ts:306]: we don't currently have other keys, but just to support future possibilities

File: src/credentials.ts (hunk 2 above)
Comment (mnzaki) [reply src/credentials.ts:105]: I have updated the API to handle `CredentialMetadataSummary` objects instead, which have an issuer profile.
`CredentialType` now gives access to an `issuerProfile` and `renderAs` and `definition`

**Pull request merged**